src/Partners/PartnersClient.php

```diff
@@ -51,16 +57,55 @@ public function getOrganizationUnitsRaw(array $parameters = []): string
      * @throws NotFoundException
      * @throws PublicApiClientException
      */
-    public function getOrganizationUnitList(array $parameters = []): Generator
+    public function getOrganizationUnits(array $parameters = []): Generator
     {
-        $rawResponse = $this->getOrganizationUnitsRaw($parameters);
-        $response = $this->getResponseData($rawResponse);
+        $this->setPerPage(100);
+        $currentPage = 1;
+        $lastPage = false;
+
+        while (! $lastPage) {
+            $this->setPage($currentPage);
+            $rawResponse = $this->getOrganizationUnitsRaw($parameters);
+            $response = $this->decodeResponse($rawResponse);
+
+            if ($response['pagination']['total_page'] <= $currentPage) {
+                $lastPage = true;
+            }
+
+            $currentPage++;
 
-        foreach ($response as $data) {
-            yield new OrganizationUnit($data);
+            foreach ($response['data']['customers'] as $data) {
+                yield new OrganizationUnit($data);
+            }
         }
     }
 
+    /**
+     * Get the units that belong to the company per page.
+     * Returns an OrganizationUnitsResponse that holds pagination information and the list of organization unit from the current page.
+     *
+     * @param array $parameters Optional parameters to add to the URL
+     *
+     * @return OrganizationUnitsResponse
+     *
+     * @throws EntityValidationException
+     * @throws GuzzleException
+     * @throws NotFoundException
+     * @throws PublicApiClientException
+     */
+    public function getOrganizationUnitPage(array $parameters = []): OrganizationUnitsResponse
+    {
+        if (! isset($this->perPage)) {
+            $this->setPerPage(100);
+        }
+
+        $rawResponse = $this->getOrganizationUnitsRaw($parameters);
+        $response[OrganizationUnitsResponse::ORGANIZATION_UNITS] = $this->getResponseData($rawResponse);
+        $response[OrganizationUnitsResponse::PAGINATION] = $this->getPagination($rawResponse);
```

Comment:
`getResponseData` and `getPagination` both use `decodeResponse` so you should not call them directly
I don't think these 2 methods should exist anyway.
Use `decodeResponse` directly:
```suggestion
        $rawResponse = $this->getOrganizationUnitsRaw($parameters);
        $decodedResponse = $this->decodeResponse($rawResponse);        
        $response = [
            OrganizationUnitsResponse::ORGANIZATION_UNITS => $decodedResponse[self::DATA],
            OrganizationUnitsResponse::PAGINATION => $decodedResponse[self::PAGINATION],
        ];
```
You could also pass directly `$decodedResponse` to your `OrganizationUnitsResponse` if you change its `const ORGANIZATION_UNITS = 'organization_units'` to `const ORGANIZATION_UNITS = 'data'`